api/client.go

```diff
@@ -428,34 +432,46 @@ func NewClient(c *Config) (*Client, error) {
 		config: c,
 	}
 
-	agentSinkPath := ""
+	// determine how to get a token
 	// order of precedence could change
+	tokenSinkPath := ""
 	switch {
 	case os.Getenv(EnvVaultToken) != "":
 		client.token = os.Getenv(EnvVaultToken)
-	case c.AgentFileSinkPath != "":
-		agentSinkPath = c.AgentFileSinkPath
-	case os.Getenv(EnvAgentFileSinkPath) != "":
-		agentSinkPath = os.Getenv(EnvAgentFileSinkPath)
-	}
-
-	// set and poll token from agent sink if it is available
-	if agentSinkPath != "" {
-		token, err := readAgentTokenFromFile(agentSinkPath)
+	case c.AgentFileSinkPath != "": // todo rename AgentFileSinkPath (could be from not an agent)
+		tokenSinkPath = c.AgentFileSinkPath
+	case os.Getenv(EnvTokenFileSinkPath) != "":
+		tokenSinkPath = os.Getenv(EnvTokenFileSinkPath)
+	case os.Getenv(EnvAgentAddr) != "":
+		tokenSinkPath = getSinkPathFromAgent(os.Getenv(EnvAgentAddr))
+	case c.AgentAddress != "":
+		tokenSinkPath = getSinkPathFromAgent(os.Getenv(c.AgentAddress))
```

Comment:
Should remove os.Getenv here.

Reply:
Err... and not assign to tokenSinkPath.

Reply:
this was some place holder in an early commit